### PR TITLE
gpu_ctx_vk: initialize the memory returned by ngli_darray_push() wher…

### PR DESCRIPTION
**Pull request by mbouron-gpsw**
…e relevant

We need to ensure that the memory contained by the different arrays
populated in create_render_resources() is properly initialized.

This commit fixes potential crashes happening in
destroy_render_resources() due to uninitialized memory left by
create_render_resources() if an error happened.

### CODE REVIEW

**Review by cboesch-gpsw — APPROVED** [review]

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]
How about `depth_stencil` in the surrounding scope?

File: libnodegl/backends/vk/gpu_ctx_vk.c [hunk 1]
```diff
@@ -191,41 +188,62 @@ static VkResult create_render_resources(struct gpu_ctx *s)
...
+            }
         }
 
-        struct rendertarget **rt = ngli_darray_push(&s_priv->rts, NULL);
-        struct rendertarget **rt_load = ngli_darray_push(&s_priv->rts_load, NULL);
-        if (!rt || !rt_load)
+        struct texture *target_color = ms_color ? ms_color : color;
+        struct texture *resolve_color = ms_color ? color : NULL;
+
+        struct rendertarget *rt;
```
Comment (cboesch-gpsw) [inline libnodegl/backends/vk/gpu_ctx_vk.c:228]: `rt = NULL` for consistency?

File: libnodegl/backends/vk/gpu_ctx_vk.c [hunk 2]
```diff
@@ -191,41 +188,62 @@ static VkResult create_render_resources(struct gpu_ctx *s)
...
+            ngli_rendertarget_vk_freep(&rt);
             return VK_ERROR_OUT_OF_HOST_MEMORY;
+        }
 
-        struct texture *color = ms_colorp ? *ms_colorp : *colorp;
-        struct texture *resolve_color = ms_colorp ? *colorp : NULL;
-        struct texture *depth_stencil = *depth_stencilp;
-        if ((res = create_rendertarget(s, color, resolve_color, depth_stencil, NGLI_LOAD_OP_CLEAR, rt)) != VK_SUCCESS ||
-            (res = create_rendertarget(s, color, resolve_color, depth_stencil, NGLI_LOAD_OP_LOAD, rt_load)) != VK_SUCCESS)
+        struct rendertarget *rt_load;
```
Comment (cboesch-gpsw) [inline libnodegl/backends/vk/gpu_ctx_vk.c:238]: ditto, `rt_load = NULL`

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]

File: libnodegl/backends/vk/gpu_ctx_vk.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/vk/gpu_ctx_vk.c]: Done.

File: libnodegl/backends/vk/gpu_ctx_vk.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/vk/gpu_ctx_vk.c]: Done.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**